utils/event.c

```diff
@@ -240,6 +240,31 @@ char * event_get_data_str(unsigned evt_id, void *data, bool verbose)
 	return str;
 }
 
+static void free_uftrace_event(struct uftrace_event *ev)
+{
+	if (!ev)
+		return;
+	free(ev->provider);
+	free(ev->event);
+	free(ev);
+}
+
+/**
+ * finish_events_file - cleanup memory for events in the given handle
+ * @handle: uftrace_data data structure
+ */
+void finish_events_file(struct uftrace_data *handle)
+{
+	struct uftrace_event *ev;
+	struct uftrace_event *prev_ev = NULL;
+
+	list_for_each_entry(ev, &handle->events, list) {
```

Comment:
Please use `list_for_each_entry_safe()` instead.  You can find some example codes as well.

Reply:
Thanks. I just updated it again.